### DISCUSSION TITLE
STM32F303 vs STM32F411

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/grblHAL#195] [General]
I've been watching the F303 port progressing so I took a look at the chip.  I do not understand why someone would choose the F303 over the F411 for a CNC motion controller.  It is more expensive and has less FLASH and RAM (by quite a lot). I see that it has some interesting peripherals (3 DACs, op amp, comparators) though do not see them as that useful for grblHAL.  Am I missing something here? Do the M4 "DSP" instructions add anything? Motion planning?

I have some thoughts about using the F303 for a couple of non-CNC projects I am considering, by the way. Glad to have looked at it.

**Comment by terjeio** [comment]
The main reason would be to use F303 Blackpill as a replacement for a F103 Red/Blue/Blackpill in an existing system? Most/none needs I2C support?

**Comment by langwadt** [comment]
they both use the same M4 core with FPU and DSP, the F3 is slower and has a different set of peripherals but none of the seem to be of any use for GRBL
